### PR TITLE
Fix wrapped assets importing their own namespace

### PR DESCRIPTION
**Pull request by mischnic**
Found as part of https://github.com/parcel-bundler/parcel/pull/7956

Right now, I've made it just replace `$x$import$y` by "module.exports" if it's a namespace (or default) self import in a wrapped asset. Not sure if it should instead maybe insert a `var $id = parcelRequire("own id")`and then return `$id`.

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 9.05s | -138.00ms |
| Cached | 472.00ms | +19.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/logo.c5bb83f1.png | 246.00b | +0.00b | 4.11s | -692.00ms 🚀 |

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>AtlasKit Editor 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 6.63s | +24.00ms |
| Cached | 276.00ms | +9.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/329492838694781523)**

File: packages/packagers/js/src/ScopeHoistingPackager.js [hunk 1]
```diff
@@ -806,7 +806,15 @@ ${code}
 
     if (imported === '*' || exportSymbol === '*' || isDefaultInterop) {
       // Resolve to the namespace object if requested or this is a CJS default interop reqiure.
-      return obj;
+      if (
+        parentAsset === resolvedAsset &&
+        this.wrappedAssets.has(resolvedAsset.id)
+      ) {
+        // Directly use module.exports for wrapped assets importing themselves.
+        return 'module.exports';
```
Comment (devongovett) [inline packages/packagers/js/src/ScopeHoistingPackager.js:814]: If the asset is wrapped, won't `obj` already be a `parcelRequire`?

Comment (mischnic) [reply packages/packagers/js/src/ScopeHoistingPackager.js:814]: For some reason, we have this condition:
https://github.com/parcel-bundler/parcel/blob/4700f42d81b2c99d7946371b05cd25f586e4a723/packages/packagers/js/src/ScopeHoistingPackager.js#L759
which will make `isWrapped` itself false in this self-import case.

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 11.82s | +60.00ms |
| Cached | 832.00ms | +300.00ms ⚠️ |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/logo.c5bb83f1.png | 246.00b | +0.00b | 5.40s | -709.00ms 🚀 |

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>AtlasKit Editor 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 8.58s | +247.00ms |
| Cached | 325.00ms | -7.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/329488650498736722)**

**Review by devongovett — APPROVED** [review]

File: packages/packagers/js/src/ScopeHoistingPackager.js (hunk 1 above)
Comment (devongovett) [reply packages/packagers/js/src/ScopeHoistingPackager.js:814]: hmm I don't remember why...

**Pull request merged**